core/variant.cpp

```diff
@@ -2309,7 +2337,24 @@ Variant::Variant(const Object *p_object) {
 	type = OBJECT;
 
 	memnew_placement(_data._mem, ObjData);
-	_get_obj().obj = const_cast<Object *>(p_object);
+
+	if (p_object) {
+
+		if (p_object->is_reference()) {
+			Reference *reference = const_cast<Reference *>(static_cast<const Reference *>(p_object));
+			if (!reference->reference()) {
```

Comment:
Shouldn't this be `init_ref()` as one can do `Variant(memnew(Reference))`?

Reply:
I think the problem is that this is used for initializing the reference, which is not supposed to be something you make within Variant.. so I wonder if we could more or less forbid it.

Reply:
or maybe I guess it would not hurt

Reply:
Whether it's meant or not to be used like that, it's possible and if that happens it will result in a bug hard to debug.